### ISSUE TITLE
[Animal][Manure] Account for Animal Time Budgeting

### ISSUE THREAD

**Issue by ew3361zh** [RuminantFarmSystems/RuFaS#2055]
The time that an animal spends in different parts of a farm should influences where manure gets deposited, how it is handled, and the associated emissions it generates under those different circumstances.

We need to find a way to account for an animal's time budget. The approach is not yet agreed upon but in the Animal-Manure connection meeting 11/4/24, it was discussed to look into:

1. Annual time budget vs daily time budget - do we need just one or do both? 
2. Should it be done by pen/animal type group/individual cow?
3. Should each different location within the time budget get its own manure stream? It was discussed in the meeting that this could end up leading to double-digit manure pools which we eventually want the code to handle but which isn't necessary. Instead, we should be thinking about how to best group location time budgets with similar manure stream compositions together before passing them to the manure module.
4. One option was thought to put the different parts of the farm into sectors and each sector would have 1 manure stream.
5. What is a good max number of streams before things get too complicated?

**Comment by matthew7838** [comment]
This issue is blocked by #2114, we will also need SME's input.

**Comment by allisterakun** [comment]
Grooming session: We will start requesting input from SMEs after the completion of phase 2 in #2092

**Comment by allisterakun** [comment]
@elle-andreen requesting some SME inputs on this.

**Comment by elle-andreen** [comment]
@allisterakun the time budget of animals by pen is being implemented with the refresh, but the larger discussion about time budgets over the year and other options to make use of them moreso in the Animal module are still open. I added a couple updates to Niko's original main discussion points below. I think there is not additional input needed at this time for manure refresh specifically - we have discussed the intended design with the animal module and external SMEs several times.

1. (SME) Annual time budget vs daily time budget - do we need just one or do both? **Daily budget by pen will be implemented when manure refresh (#2092) is closed, but annual time budget is still open for discussion.**
2. (SME/Dev) Should it be done by pen/animal type group/individual cow?
3. (SME) Should each different location within the time budget get its own manure stream? It was discussed in the meeting that this could end up leading to double-digit manure pools which we eventually want the code to handle but which isn't necessary. Instead, we should be thinking about how to best group location time budgets with similar manure stream compositions together before passing them to the manure module.
4. (Dev) One option was thought to put the different parts of the farm into sectors and each sector would have 1 manure stream.
5. (SME/Dev) What is a good max number of streams before things get too complicated? **For the time being, a maximum number of streams (from a single pen) is not enforced.**

**Comment by matthew7838** [comment]
@elle-andreen could you evaluate the priority for this issue? Also, can the above questions be categorized as SME questions or dev questions?

**Comment by elle-andreen** [comment]
@matthew7838 I would consider this low priority/nice to have. The higher priority points for the original issues scope are addressed in the manure refresh with the implementation of manure streams. I added a note of dev, SME, or both to the items above!

**Comment by matthew7838** [comment]
Grooming: Figure out what needs to be done and create a new issue.

**Comment by matthew7838** [comment]
Created a new issue for this: Issue #2520.
